presto-product-tests/src/main/java/com/facebook/presto/tests/jdbc/LdapTests.java

```diff
@@ -0,0 +1,242 @@
+    @Named("databases.presto.cli_ldap_truststore_password")
+    private String ldapTruststorePassword;
+
+    @Inject(optional = true)
+    @Named("databases.presto.cli_ldap_user_name")
+    private String ldapUserName;
+
+    @Inject(optional = true)
+    @Named("databases.presto.cli_ldap_user_password")
+    private String ldapUserPassword;
+
+    @Inject(optional = true)
+    @Named("databases.presto.host")
+    private String prestoHost;
+
+    // TODO: Find out why this doesn't bring in the value from the config file.
+    @Inject(optional = true)
+    @Named("databases.presto.ssl_port")
+    private String prestoSslPort;
+
+    private static final String INVALID_CREDENTIALS = "Invalid credentials";
+
+    @BeforeTestWithContext
+    public void setup()
+            throws SQLException
+    {
+        if (prestoSslPort == null) {
+            prestoSslPort = DEFAULT_SSL_PORT;
+        }
+
+        LOGGER.debug("Using Presto server SSL port: " + prestoSslPort);
+    }
+
+    @Override
+    public Requirement getRequirements(Configuration configuration)
+    {
+        return new LdapObjectRequirement(
+                Arrays.asList(
+                        AMERICA_ORG, ASIA_ORG,
+                        DEFAULT_GROUP, PARENT_GROUP, CHILD_GROUP,
+                        DEFAULT_GROUP_USER, PARENT_GROUP_USER, CHILD_GROUP_USER, ORPHAN_USER
+                ));
+    }
+
+    @Requires(ImmutableTpchTablesRequirements.ImmutableNationTable.class)
+    @Test(groups = {LDAP, SIMBA_JDBC, PROFILE_SPECIFIC_TESTS}, timeOut = TIMEOUT)
+    public void shouldRunQueryWithLdap()
+            throws InterruptedException, SQLException
+    {
+        assertThat(executeQuery(SELECT_QUERY, ldapUserName, ldapUserPassword)).matches(PRESTO_NATION_RESULT);
+    }
+
+    @Test(groups = {LDAP, SIMBA_JDBC, PROFILE_SPECIFIC_TESTS}, timeOut = TIMEOUT)
+    public void shouldFailQueryForLdapUserInChildGroup()
+            throws InterruptedException
+    {
+        String name = CHILD_GROUP_USER.getAttributes().get("cn");
+        expectQueryToFailForUserNotInGroup(name);
+    }
+
+    @Test(groups = {LDAP, SIMBA_JDBC, PROFILE_SPECIFIC_TESTS}, timeOut = TIMEOUT)
+    public void shouldFailQueryForLdapUserInParentGroup()
+            throws InterruptedException
+    {
+        String name = PARENT_GROUP_USER.getAttributes().get("cn");
+        expectQueryToFailForUserNotInGroup(name);
+    }
+
+    @Test(groups = {LDAP, SIMBA_JDBC, PROFILE_SPECIFIC_TESTS}, timeOut = TIMEOUT)
+    public void shouldFailQueryForOrphanLdapUser()
+            throws InterruptedException
+    {
+        String name = ORPHAN_USER.getAttributes().get("cn");
+        expectQueryToFailForUserNotInGroup(name);
+    }
+
+    @Test(groups = {LDAP, SIMBA_JDBC, PROFILE_SPECIFIC_TESTS}, timeOut = TIMEOUT)
+    public void shouldFailQueryForWrongLdapPassword()
+            throws IOException, InterruptedException
+    {
+        expectQueryToFail(ldapUserName, "wrong_password", INVALID_CREDENTIALS);
```

Comment:
could you also add credentials with special characters test? like what we have in cli tests